datafusion/common/src/dfschema.rs

```diff
@@ -28,18 +29,58 @@ use crate::{field_not_found, Column, OwnedTableReference, TableReference};
 
 use arrow::compute::can_cast_types;
 use arrow::datatypes::{DataType, Field, FieldRef, Fields, Schema, SchemaRef};
-use std::fmt::{Display, Formatter};
 
 /// A reference-counted reference to a `DFSchema`.
 pub type DFSchemaRef = Arc<DFSchema>;
 
+/// Stores identifier keys and their associated column indices. An identifier key
```

Comment:
I think a term for this concept in the database world is "Functional Dependence" where the value of one or more columns determines the value of one or more others. There is classic way to represent them and a standard set of axioms like https://en.wikipedia.org/wiki/Armstrong%27s_axioms for reasoning about them 

I recommend we change to using this the functional dependence term so that:
1. This (really cool) feature is easier to discover
2. We can rely on existing explanations of the concept more rather than having to explain it fully in DataFusion. For example, google will tell you a lot about this https://www.google.com/search?q=functional+dependence+in+database. This article seems to have a good treatment of it https://www.scaler.com/topics/dbms/functional-dependency-in-dbms/
3.  We can rely on prior art for rules and representations. In your example with `sn` and `amount`, I think would be modeled like `{sn, amount} -> {amount}`